### PR TITLE
Curating Owners: cmd/kubelet

### PR DESCRIPTION
**Pull request by apelisse**
cc @yujuhong @dchen1107 @vishh

In an effort to expand the existing pool of reviewers and establish a
two-tiered review process (first someone lgtms and then someone
experienced in the project approves), we are adding new reviewers to
existing owners files.

If You Care About the Process:
------------------------------

We did this by algorithmically figuring out who’s contributed code to
the project and in what directories.  Unfortunately, that doesn’t work
well: people that have made mechanical code changes (e.g change the
copyright header across all directories) end up as reviewers in lots of
places.

Instead of using pure commit data, we generated an excessively large
list of reviewers and pruned based on all time commit data, recent
commit data and review data (number of PRs commented on).

At this point we have a decent list of reviewers, but it needs one last
pass for fine tuning.

Also, see https://github.com/kubernetes/contrib/issues/1389.

TLDR:
-----

As an owner of a sig/directory and a leader of the project, here’s what
we need from you:

1. Use PR https://github.com/kubernetes/kubernetes/pull/35715 as an example.

2. The pull-request is made editable, please edit the `OWNERS` file to
remove the names of people that shouldn't be reviewing code in the future in
the **reviewers** section. You probably do NOT need to modify the **approvers**
section. Names asre sorted by relevance, using some secret statistics.

3. Notify me if you want some OWNERS file to be removed.  Being an
approver or reviewer of a parent directory makes you a reviewer/approver
of the subdirectories too, so not all OWNERS files may be necessary.

4. Please use ALIAS if you want to use the same list of people over and
over again (don't hesitate to ask me for help, or use the pull-request
above as an example)

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/36394)

### CODE REVIEW

**Comment by apelisse** [comment]
Please consider removing the `kubelet/app/OWNERS` if it sounds redundant.

**Comment by vishh** [comment]
@apelisse I went ahead and pruned the list for now. There are quite a few active contributors missing from the pre-existing list, but that OK since we can add more people once this framework lands.

**Comment by apelisse** [comment]
You are more than welcome to:
- tell me who our imperfect process might have missed,
- add names in the list if needed

Thank you very much for reviewing the list!
On Tue, Nov 8, 2016, 5:22 PM Vish Kannan <notifications@github.com> wrote:

@apelisse <https://github.com/apelisse> I went ahead and pruned the list
for now. There are quite a few active contributors missing from the
pre-existing list, but that OK since we can add more people once this
framework lands.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/kubernetes/kubernetes/pull/36394#issuecomment-259311091>,
or mute the thread

**Comment by apelisse** [comment]
Hi @vishh, any interest in creating an alias for these people so that it can be reused more easily? sig-node?

**Comment by vishh** [comment]
I'm not sure if an alias is a good idea. Having IDs explicitly listed will
allow devs to post patches to add or remove themselves instead of
requesting org admins to perform those tasks.

**Comment by yujuhong** [comment]
@vishh, I think @apelisse means adding a sig-node list in https://github.com/kubernetes/kubernetes/blob/master/OWNERS_ALIASES, where people can still add/remove through PRs. The advantage is that you can reuse the same list for multiple packages.

**Comment by apelisse** [comment]
Let's merge as it is. We can use "alias" later if needed. We need to keep this moving forward.

**Comment by dchen1107** [comment]
Did we agree the reviewer should be sig-node-reviewer mailing list in our discussion a while back?

**Comment by calebamiles** [comment]
@vishh would you be fine with 

a) merging this PR as even though the lack of a reviewer alias will make this OWNERS file inconsistent with the ones in:
- https://github.com/kubernetes/kubernetes/pull/36467
- https://github.com/kubernetes/kubernetes/pull/36528

b) update this PR to us a reviewer alias with the idea that your desire to allow developers to submit PRs to add or remove themselves will be taken into account in the final location of the `OWNERS_ALIAS` file (it might make some sense to have the list of reviewers collocated with SIG information in `kubernetes:community` or similar)

c) merge this PR and immediately follow with a PR to use a reviewer alias while keeping the considerations of option (b)

I would really like to see some action on this PR since this relatively simple process has been going on for quite some time and we haven't gotten to the trickier part of deciding how someone becomes an approver after being a reviewer which is something that we need to hash out as a community in a different thread.

**cc: @apelisse, @dchen1107, @yujuhong**

File: cmd/kubelet/OWNERS
Comment (vishh) [inline cmd/kubelet/OWNERS]: This list can be `sig-node-reviewers`.

File: cmd/kubelet/app/OWNERS
Comment (vishh) [inline cmd/kubelet/app/OWNERS]: Ditto here

File: cmd/kubelet/OWNERS
Comment (vishh) [inline cmd/kubelet/OWNERS]: Add `derekwaynecarr` and `Random-Liu`

**Review by vishh — CHANGES_REQUESTED** [review]
Two basic comments. Let's get this through.

**Comment by calebamiles** [comment]
Thanks for the review @vishh. @apelisse would you like to make the changes? alternatively you could make me a collaborator on your fork and I'm happy to make these small changes.

**Comment by calebamiles** [comment]
comments should be addressed @vishh.


**cc: @apelisse, @dchen1107**

**Review by vishh — APPROVED** [review]

**Pull request merged**
